### PR TITLE
Fix LLVM 5.0 build

### PR DESCRIPTION
**Pull request by JohanEngelen**
Quite a significant change here: https://reviews.llvm.org/D21723

Shall we also update the identifiers in the SyncScopeID struct for LLVM>=5.0?
https://github.com/ldc-developers/druntime/blob/cc9335131218cf157af7e562309b64e5b0c51bb3/src/ldc/intrinsics.di#L357-L361

### CODE REVIEW

**Comment by JohanEngelen** [comment]
@thewilsonator Another update required for the SPIRV build!

**Comment by thewilsonator** [comment]
New one up, hot off my laptop.

**Comment by thewilsonator** [comment]
> Quite a significant change here: https://reviews.llvm.org/D21723

That looks like a very interesting change very applicable for SPIRV. Will have to take a closer look.

>Shall we also update the identifiers in the SyncScopeID struct for LLVM>=5.0?

Yeah that would be great.

File: gen/tocall.cpp [hunk 1]
```diff
@@ -332,11 +332,17 @@ bool DtoLowerMagicIntrinsic(IRState *p, FuncDeclaration *fndecl, CallExp *e,
     }
     auto atomicOrdering =
         static_cast<llvm::AtomicOrdering>((*e->arguments)[0]->toInteger());
+#if LDC_LLVM_VER >= 500
+    llvm::SyncScope::ID scope = (e->arguments->dim == 2)
+                                    ? (*e->arguments)[1]->toInteger()
+                                    : llvm::SyncScope::System;
+#else
```
Comment (thewilsonator) [inline gen/tocall.cpp]: I think you need to `#include "llvm/IR/LLVMContext.h"` at the top of this file so that `llvm::SyncScope` is in scope.

**Comment by thewilsonator** [comment]
Other than the missing header and assuming the tests work, LGTM.

**Comment by kinke** [comment]
It's not Circle's fault, it's the [LLVM APT repo](http://apt.llvm.org/) which only provides frequent updates (something like daily) for the 3 latest Ubuntu versions (16.04, 16.10 and 17.04) and only seldom updates for the older ones (trusty: June 21st, which is actually quite good, compared to precise with April 12th (!)).

**Comment by kinke** [comment]
It may pay off to look into CircleCI's docker support and use a Ubuntu 16.04 image for it, so that its LLVM 5.0 is always up to speed.

**Pull request merged**